_posts/2020-11-05-data-visualization-in-R.md

```diff
@@ -0,0 +1,141 @@
+### Example  
+Unlike bar charts where the independent variable may go on either the x-axis or y-axis, line graphs exclusively place the independent variable on the x-axis.  
+![line graph](https://ggplot2.tidyverse.org/reference/geom_path-1.png)
+
+## Box Plots  
+Box plots are very useful when a scientists is trying to demonstrate distributions. These plots show the density distribution, the mean, max, and minimum. Additionally, customization allows the user to exclude outliers in these calculations while still including them on the plot. A common alternative to a box plot is a violin plot.  
+### Example
+Box plots and violin plots can be customized to include or exclude outliers when representing the density distribution.
+![Box plot showing outliers](https://ggplot2.tidyverse.org/reference/geom_boxplot-1.png)
+
+## Scatter/Point Plots  
+Scatter plots are optimal for data that you have many data points for which an average does not make sense but you would like to highlight specific grouping or trends.
+### Example
+![Classic scatter plot with categories](https://ggplot2.tidyverse.org/reference/geom_point-2.png)
+
+# Using  `ggplot`  
+`ggplot` is a data visualization system from tidyverse that takes input data sets and creates customizable graphics. For information beyond this post and user guides, github provides a [ggplot overview](https://ggplot2.tidyverse.org/index.html). Prior to running any script you will have to load the library for  `ggplot` or `ggplot2`.  Note that this is different than installing the packages.
+
+To do this: ```library(ggplot2)```
+
+## Writing a script and data input
+The general format of a ggplot input is as follows:
+```
+{r name_of_plot}
+name_of_plot <- ggplot(data = data_source,
+                       aes(x = geo_type,
+                           y = n,
+                           fill = actual_data_to_plot)) +
+  geom_col(position = position_dodge()) +
+  labs(title = paste("Your title here",
+                     dynamic_input),
+       x = "Type of area",
+       y = "Number of areas")
+```
+
+Saving a ggplot to a file:
+```
+ggsave(plot = name_of_plot,
+       filename = paste0("data directory",
+                         core_name,
+                         "file_name.png"))
+```
+
+Finally, in R, if you want to see the generated figure without opening the saved file, you just call the plot name at the end of the script:
+```
+name_of_plot
+```
+
+Interpreting the script above based on inputs and commands:  
+1. `{r name_of_plot}`  
+  * In this command we are calling for R to create a plot with this specific title
+2. `data = `
+  * We have to read in the data file, the easiest way to do this is from a csv or txt file and prior to creating the figure set a variable equal to this file. This way you can call the variable and that will read in the data as opposed to giving the precise location of the data each time.
```

Comment:
Not exactly. When the command is run to create the data object, it reads from the file and stores the data into memory. That happens when the variable is created, not when the ggplot call occurs. So maybe a slight rewording here would be good.